BC-MAR-21-ROR/kata-07-rejoj-checador-equipo-03-k07
# Feature#11 check in and check out

# Description

Add page for employees to check_in and check_out using their private number in the input screen to track attendance, and in the future this will be used to generate reports.

When an Employee checks in:
 - Log record is created with the time of check_in. 
 - Message is shown to the employee explaining if the check_in was successful or failed.

When an Employee checks out: 
 - Log record is updated with the time of check_out.
 - Message is shown to the employee explaining if the check_out was successful or failed.

There are validations to only allow one check_in/check_out by day per employee.
There are tests that cover this feature.

See issue #11 for more information.
 
 # Testing
 - rails db:migrate
 - rails db:seed:replant 
 - Grab an employee's private number from database using `rails console` :
   ```ruby
   Employee.pluck :private_number
   ```
 - Run `rails server`
 - Run `bin/webpack-dev-server`
 - Navigate to: `http://localhost:3000/logs/check_in` enter a valid private number to check_in
 - Now, in`http://localhost:3000/logs/check_out` enter the same private number to check_out
 - Employee should see visual feedback
   
  ## Screenshots 

![Screen Recording 2021-04-21 at 08 45 46 p m](https://user-images.githubusercontent.com/11895273/115610475-31b5a680-a2e9-11eb-8797-6ae3b46deb2e.gif)

### PSQL

![Image 2021-04-21 at 9 26 36 p m](https://user-images.githubusercontent.com/11895273/115611249-2020ce80-a2ea-11eb-844e-42621c78bb78.jpg)


Co-authored-by: EusebioAjas <eusebioajas6@gmail.com>
Co-authored-by: RobertVel666 <robertvelasco99@gmail.com>
Co-authored-by: SpidySamurai <javierchiortiz@gmail.com>
